### PR TITLE
Port avatar to react

### PR DESCRIPTION
**Pull request by campos20**
This was opened so another person can continue it

### CODE REVIEW

**Comment by dunkOnIT** [comment]
Thanks Alexandre!

**Comment by gregorbg** [comment]
I'll be handling the merge conflicts and building a backend for the React components

File: infra/wca_on_rails/staging/s3.tf [hunk 1]
```diff
@@ -10,3 +10,9 @@ resource "aws_s3_bucket" "avatars" {
     "Name" = "wca-avatar"
   }
 }
+resource "aws_s3_bucket" "avatars_private" {
```
Comment (FinnIckler) [inline infra/wca_on_rails/staging/s3.tf:13]: you are declaring the bucket twice, if you want staging and production to have access to the bucket you need to create it in the shared folder, declare it as an output (and in variables.tf under the shared module) and use that output in the staging and production files. I can do it if you want!

Comment (gregorbg) [reply infra/wca_on_rails/staging/s3.tf:13]: I would actually like to take you up on this offer. TIA :smile:

Comment (FinnIckler) [reply infra/wca_on_rails/staging/s3.tf:13]: I'll do it now

**Comment by FinnIckler** [comment]
Before I run terraform apply, should staging have access to the private bucket? If yes, do you have any safeguards in place that users can't somehow replace their avatars there?

**Comment by gregorbg** [comment]
> Before I run terraform apply, should staging have access to the private bucket? If yes, do you have any safeguards in place that users can't somehow replace their avatars there?

There is no "smart" safeguard in place, but the `AVATARS_PUBLIC_STORAGE` and `AVATARS_PRIVATE_STORAGE` variables can control which storage backends (as defined in `config/storage.yml`) the server even tries to read from.

Can you configure the infrastructure in such a way that (a) staging gets its own private bucket and (b) the access from staging to the public bucket is read-only?

**Comment by FinnIckler** [comment]
done!

**Comment by FinnIckler** [comment]
I can run terraform whenever

**Comment by gregorbg** [comment]
@thewca-bot deploy staging

**Comment by gregorbg** [comment]
This has grown too big over time to demand a review from anybody.
Merging this on my authority.

**Comment by maxidragon** [comment]
Default avatars seem to be broken. See example https://www.worldcubeassociation.org/persons/2023ROGA01

**Comment by FinnIckler** [comment]
already a fix open at https://github.com/thewca/worldcubeassociation.org/pull/10083

**Pull request merged**